frontend/src/pages/NewRun.tsx

```diff
@@ -136,59 +157,106 @@ class NewRun extends Page<{}, NewRunState> {
                       style={{ padding: '3px 5px', margin: 0 }}>
                       Choose
-                </Button>
+                    </Button>
                   </InputAdornment>
                 ),
                 readOnly: true,
               }} />
           )}
 
           <Dialog open={pipelineSelectorOpen}
-            classes={{ paper: css.pipelineSelectorDialog }}
+            classes={{ paper: css.selectorDialog }}
             onClose={() => this._pipelineSelectorClosed(false)}
             PaperProps={{ id: 'pipelineSelectorDialog' }}>
             <DialogContent>
-              <PipelineSelector {...this.props} pipelineSelectionChanged={this._pipelineSelectionChanged.bind(this)} />
+              <ResourceSelector {...this.props}
+                title='Choose a pipeline'
+                listApi={async (...args) => {
+                  const response = await Apis.pipelineServiceApi.listPipelines(...args);
+                  return { resources: response.pipelines || [], nextPageToken: response.next_page_token || '' };
+                }}
+                columns={this.pipelineSelectorColumns}
+                emptyMessage='No pipelines found. Upload a pipeline and then try again.'
+                initialSortColumn={PipelineSortKeys.CREATED_AT}
+                resourceToRow={this._resourceToRow}
+                selectionChanged={(selectedPipeline: ApiPipeline) =>
+                  this.setStateSafe({ unconfirmedSelectedPipeline: selectedPipeline })}/>
             </DialogContent>
             <DialogActions>
               <Button id='cancelPipelineSelectionBtn' onClick={() => this._pipelineSelectorClosed(false)} color='secondary'>
                 Cancel
               </Button>
               <Button id='usePipelineBtn' onClick={() => this._pipelineSelectorClosed(true)}
-                color='secondary' disabled={!unconfirmedDialogPipelineId}>
+                color='secondary' disabled={!unconfirmedSelectedPipeline}>
                 Use this pipeline
               </Button>
             </DialogActions>
           </Dialog>
 
+          <Dialog open={experimentSelectorOpen}
+            classes={{ paper: css.selectorDialog }}
+            onClose={() => this._experimentSelectorClosed(false)}
+            PaperProps={{ id: 'experimentSelectorDialog' }}>
+            <DialogContent>
+            <ResourceSelector {...this.props}
+                title='Choose an experiment'
+                listApi={async (...args) => {
+                  const response = await Apis.experimentServiceApi.listExperiment(...args);
+                  return { resources: response.experiments || [], nextPageToken: response.next_page_token || '' };
+                }}
+                columns={this.experimentSelectorColumns}
+                emptyMessage='No experiments found. Create an experiment and then try again.'
+                initialSortColumn={ExperimentSortKeys.CREATED_AT}
+                resourceToRow={this._resourceToRow}
```

Comment:
If both components use the same method, why not integrate it for now, until we need to separate it out?